### PR TITLE
Addition of names of planets along with sun.

### PR DESCRIPTION
**Pull request by ghost**
Add Names to the Planets and sun.
On landing / homescreen page of activity this adds names to the planets and sun.

### CODE REVIEW

**Comment by ghost** [comment]
Kindly review @chimosky @walterbender @quozl

File: area.py [hunk 1]
```diff
@@ -203,12 +235,13 @@ def get_all_bodies(self):
...
 
     def get_body_radius(self, body):
-        if body.name == BodyName.SUN:
-            return get_sun_scale_radius(self.width, self.height, self.zoom)
+            if body.name == BodyName.SUN:
+                return get_sun_scale_radius(self.width, self.height, self.zoom)
+
+            else:
+                return get_planet_scale_radius(
+                    self.width, self.height, body, self.zoom)
```
Comment (chimosky) [inline area.py]: Why the extra indentation?

**Comment by chimosky** [comment]
Reviewed 4562a56, not tested.

File: area.py (hunk 1 above)
Comment (ghost) [reply area.py]: i corrected it also , on Zed new IDE its happening sorry for it ,you can check latest, i corrected it that time only.

**Comment by ghost** [comment]
Already Addressed @chimosky you can test final changes.

**Comment by ghost** [comment]
kindly test this once if you can @quozl.

**Comment by quozl** [comment]
Tested.

Note low commit message quality and code style makes flake8 worse;

```diff
--- master branch
+++ pull request branch
@@ -1,8 +1,13 @@
 ./area.py:62:57: W504 line break after binary operator
 ./area.py:63:59: W504 line break after binary operator
 ./area.py:64:58: W504 line break after binary operator
-./area.py:156:61: W504 line break after binary operator
-./area.py:190:53: W504 line break after binary operator
+./area.py:157:61: W504 line break after binary operator
+./area.py:174:9: E303 too many blank lines (2)
+./area.py:189:53: W504 line break after binary operator
+./area.py:193:80: E501 line too long (91 > 79 characters)
+./area.py:198:9: E303 too many blank lines (3)
+./area.py:212:53: W504 line break after binary operator
+./area.py:247:5: E303 too many blank lines (2)
 ./toolbars.py:382:78: W504 line break after binary operator
 ./toolbars.py:383:80: E501 line too long (90 > 79 characters)
 ./toolbars.py:383:90: W504 line break after binary operator
```

It would be great not to go backwards.

**Comment by ghost** [comment]
Well if forced pushed commit message are of no good, then is it possible for you to give me a example, by taking this particular case how would you have written the commit message it will really help me and also I have read about imperative mood and one line summary and changing branch from sugar-docs for contributor.
Also Blank lines are the  problems I will remove them Thankyou.

**Comment by ghost** [comment]
Also more regarding coding style is there any problem with variable name chosen/ or anything else other than extra blank line.

**Comment by ghost** [comment]
I have removed extra lines and refactored line breaks to ensure compliance with flake8 standards. Thankyou.
Also the latest commit message was of any good, at this point of time I am really confused how one has to write them😢.

**Comment by chimosky** [comment]
> I have removed extra lines and refactored line breaks to ensure compliance with flake8 standards. Thankyou. Also the latest commit message was of any good, at this point of time I am really confused how one has to write them😢.

For your first commit, a better commit message could be in the form;
```
Enhance visualization - summary of commit - 

Add names for planets and sun, optimize presentation... - details of commit - 
```

I added the _summary of commit_ and _details of commit_ to give you an idea of what the commit message should look like when a one line summary isn't enough, the details should state what was done and if there's more to do it should also state that, you can use a TODO list if needed.
The details also explain your reasoning sometimes if your changes might introduce any ambiguity.

**Comment by ghost** [comment]
Thankyou very much for the detailed explanation let me try again.

**Comment by ghost** [comment]
kindly review and see the commit message now. @chimosky @quozl @walterbender.

**Comment by quozl** [comment]
https://github.com/sugarlabs/solar-system/pull/8/commits/2d05eca0a2c92944ac64e22ad308cb93b9058df6 is how I'm looking at it first.

* [ ] the first line exceeds the recommended length, which is why GitHub has split it,
* [ ] the body lines are also exceeding recommended length, and while GitHub does wrap them, git itself does not;

```log
$ git fetch origin pull/8/head
$ git log 2d05ec
commit 2d05eca0a2c92944ac64e22ad308cb93b9058df6
Author: Anurag Singh <principalquantum30@gmail.com>
Date:   Mon Mar 18 03:48:42 2024 +0530

    Enhance visualization: Add names for planets and sun, optimize presentation.
    
    - Implemented text labels for planets and the Sun to improve identification of celestial bodies
    - Dynamically positioned planet names next to corresponding bodies for proper alignment and readability
    - Added green hover circle to provide visual feedback during interaction
    -Refactor line breaks and remove blank lines for improved code cleanliness
```

Which looks like this in Terminal;

git

The body text of the commit message is a bit hard to parse.

But it does finally bring up a point, now that I understand your intention better by reading your commit message; the activities are meant to be discoverable, and the lack of label for objects was part of the original design, to encourage children to click on the objects for more detail.  By adding a label, you've de-emphasised this part of the design, and it isn't so easily discoverable.  Also, it doesn't necessarily help with identification to add a label.

The labels are also only in English.  The activity hasn't been internationalised.  The Planets activity in Sugarizer may be a better focus, it at least has internationalisation support,  https://try.sugarizer.org/activities/Planets.activity/index.html

**Comment by ghost** [comment]
oh ok it was intentional not to mention name, ok will close this pr, Thanks.

**Comment by quozl** [comment]
Well, _I_ could be wrong.  What do you think, @chimosky, @walterbender?

**Comment by walterbender** [comment]
I don't think having the names in the orbit view takes away from the learning experience. And maybe the label could be clickable (for me at least, clicking on the smaller planets can be challenging).

**Comment by ghost** [comment]
If all the devs/mentors agree:

- [ ] Making text a little bigger not too much
- [ ] Making text label clickable
- [ ] Internationalisation


Not a problem but with internationalisation another task come as a burden to make the information written about planets also in multiple language but I can try thankyou.

**Comment by chimosky** [comment]
I see Quozl's point, but I agree with Walter, the text being clickable also helps.

Internationalization for this activity would be a great way to test weblate, I haven't gotten
around to creating a github workflow that'll automatically open PRs.

PS: Drop the mentors/dev tag as you can communicate without it.

**Comment by quozl** [comment]
> Not a problem but with internationalisation another task come as a burden to make the information written about planets also in multiple language but I can try thankyou.

Not so.  Set up the internationalisation in the code, make the `po` directory, generate the POT file, add any languages that _you_ are fluent in, but leave the rest for others.  Don't attempt machine translation.

**Comment by chimosky** [comment]
> If all the devs/mentors agree:
> 
>     * [ ]  Making text a little bigger not too much
> 
>     * [ ]  Making text label clickable
> 
>     * [ ]  Internationalisation
> 
> 
> Not a problem but with internationalisation another task come as a burden to make the information written about planets also in multiple language but I can try thankyou.

We agreed on this checklist.

**Comment by ghost** [comment]
Yes, thanks will start working on it soon!!

**Comment by ghost** [comment]
> [2d05eca](https://github.com/sugarlabs/solar-system/commit/2d05eca0a2c92944ac64e22ad308cb93b9058df6) is how I'm looking at it first.
> 
> * [ ]  the first line exceeds the recommended length, which is why GitHub has split it,
> * [ ]  the body lines are also exceeding recommended length, and while GitHub does wrap them, git itself does not;
> 
> ```
> $ git fetch origin pull/8/head
> $ git log 2d05ec
> commit 2d05eca0a2c92944ac64e22ad308cb93b9058df6
> Author: Anurag Singh <principalquantum30@gmail.com>
> Date:   Mon Mar 18 03:48:42 2024 +0530
> 
>     Enhance visualization: Add names for planets and sun, optimize presentation.
>     
>     - Implemented text labels for planets and the Sun to improve identification of celestial bodies
>     - Dynamically positioned planet names next to corresponding bodies for proper alignment and readability
>     - Added green hover circle to provide visual feedback during interaction
>     -Refactor line breaks and remove blank lines for improved code cleanliness
> ```
> 
> Which looks like this in Terminal;
> 
> git
> 
> The body text of the commit message is a bit hard to parse.
> 
> But it does finally bring up a point, now that I understand your intention better by reading your commit message; the activities are meant to be discoverable, and the lack of label for objects was part of the original design, to encourage children to click on the objects for more detail. By adding a label, you've de-emphasised this part of the design, and it isn't so easily discoverable. Also, it doesn't necessarily help with identification to add a label.
> 
> The labels are also only in English. The activity hasn't been internationalised. The Planets activity in Sugarizer may be a better focus, it at least has internationalisation support, https://try.sugarizer.org/activities/Planets.activity/index.html

Here as @quozl said it turns out to be true,I tried with the changes, everything seems oversimplified which makes this activity rather boring, then to encourage them for more details.
I think there might be any better idea to achieve this goal without compromising discoverability of this activity.
Thanks.
